proposals/lip-0053.md

```diff
@@ -252,178 +479,192 @@ crossChainUpdateTransactionParams = {
+            ccmID = SHA-256(ccmBytes)
+            
+            emitEvent(
+                moduleID = MODULE_ID_INTEROPERABILITY,
+                typeID = TYPE_ID_CCM_PROCESSED,
+                data = {"ccmID": ccmID, "status": CCM_PROCESSED_STATUS_COMMAND_DISCARDED},
+                topics = [ccm.sendingChainID, ccm.receivingChainID]
+            )
+            # Notice that, even if the sending chain has been terminated, 
+            # future CCMs from other chains can still be processed.
+            continue
+        
+        apply(trs, ccm)
+
+        # Finally update the inbox
+        appendToInboxTree(trs.params.sendingChainID, ccmBytes)
+    
+    # Update mainchain validators.
+    updateValidators(trs)
+    
+    # Update last certificate.
+    if len(trs.params.certificate) > 0:
+        updateLastCertificate(trs)
+    
+    # Update partnerChainOutboxRoot.
+    updatePartnerChainOutboxRoot(trs)
+```
 
-* Set `partnerValidators.activeValidators = updateActiveValidators(partnerValidators.activeValidators, activeValidatorsUpdate)` as specified in the ["Update Active Validators" section](#update-active-validators).
-* If `newCertificateThreshold != 0`, update `partnerValidators.certificateThreshold` to `newCertificateThreshold`.
-* If `params` contains a non-empty `certificate`:
-  * Set `partnerChain.lastCertificate.stateRoot` to `certificate.stateRoot`.
-  * Set `partnerChain.lastCertificate.timestamp` to `certificate.timestamp`.
-  * Set `partnerChain.lastCertificate.height` to `certificate.height`.
-  * Set `partnerChain.lastCertificate.validatorsHash` to `certificate.validatorsHash`.
-* Set `partnerChannel.partnerChainOutboxRoot` to:
-  * `calculateRootFromRightWitness(inbox.size, inbox.appendPath, inboxUpdate.messageWitness.siblingHashes)` if `params` contains a non-empty `inboxUpdate.messageWitness`, this function is defined in [LIP 0031][lip-0031#rightWitness].
-  * `partnerChannel.inbox.root` otherwise.
+### Cross-chain Update Processing
 
-### Update Active Validators
+### Verify Cross-chain Message
 
-Updating `partnerValidators.activeValidators` with respect to a given `activeValidatorsUpdate` is done following the logic below:
+During the verification of a cross-chain message `ccm`, the following logic is applied:
 
 ```python
-updateActiveValidators(activeValidators, activeValidatorsUpdate)
-    for validator in activeValidatorsUpdate:
-        if there exist currentValidator in activeValidators such that
-        validators.blsKey == currentValidator.blsKey:
-            set currentValidator.bftWeight = validators.bftWeight
-        else:
-            add validator to activeValidators maintaining the entries in
-            lexicographical order of blsKeys
-
-    remove any entry activeValidators with bftWeight == 0
-    return activeValidators
+def verifyCrossChainMessage(ccm: CCM) -> None:
+    # If the sending chain has been terminated by any previous check, interrupt the CCU processing.
```

Comment:
I don't understand this comment? In the execute logic above the CCU processing continues.

Reply:
It was a remnant of refactoring. I removed the comments

Reply:
Can be resolved then.